cli/flox-rust-sdk/src/models/environment/generations.rs

```diff
@@ -571,6 +574,76 @@ pub struct GenerationId(usize);
 
 #[cfg(test)]
 mod tests {
-    // todo: tests for this will be easier with the `init` method implemented
-    // in https://github.com/flox/flox/pull/563
+    use super::*;
+    use crate::flox::test_helpers::flox_instance;
+    use crate::models::environment::Environment;
+    use crate::models::environment::path_environment::test_helpers::new_path_environment;
+
+    #[test]
+    fn test_set_current_generation() {
+        let (flox, tempdir) = flox_instance();
+        let env = new_path_environment(&flox, "version = 1");
+        let env_name = env.name();
+        let mut core_env = env.into_core_environment().unwrap();
+
+        let floxmeta_checkedout_path = tempfile::tempdir_in(&tempdir).unwrap().keep();
+        let floxmeta_temp_path = tempfile::tempdir_in(&tempdir).unwrap().keep();
+
+        let mut generations = Generations::init(
+            GitCommandOptions::default(),
+            floxmeta_checkedout_path,
+            floxmeta_temp_path,
+            "some-branch".to_string(),
+            &env_name,
+        )
+        .unwrap();
+
+        let first_gen_id = GenerationId(1);
+        let second_gen_id = GenerationId(2);
+        let mut generations_rw = generations.writable(&tempdir).unwrap();
+        generations_rw
+            .add_generation(&mut core_env, "First generation".to_string())
+            .unwrap();
+        generations_rw
+            .add_generation(&mut core_env, "Second generation".to_string())
+            .unwrap();
+        assert_eq!(
+            generations_rw.metadata().unwrap().current_gen,
+            Some(second_gen_id.clone()),
+            "should be at second generation"
+        );
+
+        generations_rw
+            .set_current_generation(first_gen_id.clone())
+            .unwrap();
+        assert_eq!(
+            generations_rw.metadata().unwrap().current_gen,
+            Some(first_gen_id),
+            "should roll back to first generation"
+        );
+
+        generations_rw
+            .set_current_generation(second_gen_id.clone())
+            .unwrap();
+        assert_eq!(
+            generations_rw.metadata().unwrap().current_gen,
+            Some(second_gen_id),
+            "should roll forwards to second generation"
+        );
+
+        let res = generations_rw.set_current_generation(GenerationId(10));
+        assert!(
+            matches!(res, Err(GenerationsError::GenerationNotFound(10))),
+            "should error when setting nonexistent generation, got: {:?}",
+            res
+        );
+
+        let current_gen = generations_rw.metadata().unwrap().current_gen.unwrap();
+        let res = generations_rw.set_current_generation(current_gen);
+        assert!(
+            matches!(res, Err(GenerationsError::RollbackToCurrentGeneration)),
+            "should error when setting current generation, got: {:?}",
+            res
+        );
```

Comment:
I think i'd still prefers separate tests for these if we can make the the preparatory steps more ergonomic.